### DISCUSSION TITLE
Login logs.

### DISCUSSION THREAD

**Discussion by electr1chka** [Labs64/NetLicensing-Community#136] [Questions] [answered]
My client is having trouble logging in. He is worried that having an api key, a module number and a product, anyone will be able to download a licensed product. Therefore, a check on the host is necessary. Or logging user requests.

**Comment by r-brown** [comment, marked as answer]
Hello @electr1chka 

You may need to consider following activities to improve security:
- enabling response signing as it is described here https://netlicensing.io/wiki/signing-the-response
- specify concrete license**e** while creating API key - this will restrict usage of the given API Key to the one concrete licensee only
- use minimal API key role `ROLE_APIKEY_LICENSEE ` to restrict API usage; see more at https://netlicensing.io/wiki/security#api-key-identification

**Comment by electr1chka** [comment]
What about requests logs? ip, url, timestamp?

**Reply by r-brown** [reply]
> What about requests logs? ip, url, timestamp?

Request logs are not exposed via API and therefore cannot be accessed via unauthorized channel.
